common/interpreter/error.go

```diff
@@ -18,16 +17,22 @@ type FaultyResponseHandler interface {
 
 // ErrorHandler invokes a function that matches response media type with parse error, ex: JSON<->JsonParserMethod
 // otherwise defaults to general error interpretation.
+// UnknownMedia is a special handler that is used in case Content-Type is unavailable.
+// Fallback on the other side is such a handler that will be invoked when media type is known but no handler exists.
 type ErrorHandler struct {
-	JSON FaultyResponseHandler
-	XML  FaultyResponseHandler
-	HTML FaultyResponseHandler
+	JSON         FaultyResponseHandler
+	XML          FaultyResponseHandler
+	HTML         FaultyResponseHandler
+	UnknownMedia FaultyResponseHandler
+	Fallback     FaultyResponseHandler
 }
 
-func (h ErrorHandler) Handle(res *http.Response, body []byte) error {
+func (h ErrorHandler) Handle(res *http.Response, body []byte) error { // nolint:cyclop
 	mediaType, _, err := mime.ParseMediaType(res.Header.Get("Content-Type"))
-	if err != nil {
-		return fmt.Errorf("%w: %w", ErrMissingContentType, err)
+	if err != nil && h.UnknownMedia != nil {
```

Comment:
This essentially is just adding another layer of complexity in my opinion. Why are we

Reply:
This is essentially requires the handler to accept another method for UnknowMedia, and Fallback. This does not prevent the error messages to be wrongly parsed when there is no specific handlers.

Reply:
It provides customization to the deep connector. Salesforce is so far the only one that needs to handle absent media type in error response. How is this handled is up to the Salesforce package.

Let me explain, the path how we ended up here. Consider the base case, where all error response for all connectors are parsed using `common.InterpretError(res, body)`. What it does is use the whole body as error text and enhance with some trivial errors based on status code.
In most cases we will get 400 error like such:
``
error caller: Arbitrary text from response body
``

To improve this we can stop treating error response as `opaque bytes`. To do this we need to know the language.
Most are JSON, sometimes HTML may popup in the inconsitent APIs. Thats why media type is checked to judge if we can reason about `opaque string` in a valuable manner.

For Salesforce I would imagine a separate PR just to specify how unknown media should be handled. 
Otherwise, it will be treated as opaque string via `common.InterpretError(res, body)`, thats how all errors were processed long time ago and a present fallback.

I think `Fallback` method shouldn't really be used, so that can be removed. Ideally, new media should be added as a property to error handler.

Reply:
I think in this case, we can have one function that returns error type `UnparseableError` which will include raw responses. If there is no mimetype or fails to get mimetype. But let's not make it method. We can do something like 

```
if err != nil || mediaType == "" {
  return common.HandleUnparseableHTTPResponseErrror(res, body) // return type is common.UnparseableHTTPResponseError 
}
```

so that we don't have to attach define `UnknownMedia` every time

Reply:
cc @eberle1080

Reply:
I cannot replicate the error because the script needs some configuring which is hard to know what parts are common and what unique per salesfroce account.
![Uploading image.png…]()

Special error wrapper makes sense. I will add a commit attempting to do this.

Reply:
With or without the changes in the current PR it seems that I can run the `event-flow` without any issues 
![image](https://github.com/user-attachments/assets/9acf8904-e032-4899-9dd4-6d745a4616f3)

Since Salesforce event flow script doesn't return any errors it doesn't make the change about missing media type irrelevant. Therefore, I added mock tests to demonstrate how error handler would behave.

Related changes are grouped into 3 commits.